### PR TITLE
Migrate formatting middlewares to ``Method`` formatters

### PR DESCRIPTION
**Pull request by dylanjw**
### What was wrong?

Based on: #1166

Related to Issue #1161 and #1057

TODO:

- [x] pythonic_middleware
- [x] attrdict_middleware middleware
- [x] request_param_normalizer middleware
- [x] ~validation_middleware~
- [x] abi_middleware

#### Cute Animal Picture

image

### CODE REVIEW

**Comment by pipermerriam** [comment]
waiting for #1166 to land before looking this over.

File: web3/manager.py [hunk 1]
```diff
@@ -60,14 +75,14 @@ def default_middlewares(web3):
...
-            (abi_middleware, 'abi'),
+            (request_parameter_normalizer, 'request_param_normalizer'),  # Delete
+            (gas_price_strategy_middleware, 'gas_price_strategy'),  # Add Async
+            (name_to_address_middleware(web3), 'name_to_address'),  # Add Async
+            (attrdict_middleware, 'attrdict'),  # Delete
+            (pythonic_middleware, 'pythonic'),  # Delete
+            (normalize_errors_middleware, 'normalize_errors'),  # Add async
+            (validation_middleware, 'validation'),  # Add async
+            (abi_middleware, 'abi'),  # Delete
         ]
```
Comment (dylanjw) [inline web3/manager.py]: These comments are notes to myself.  Will remove.

**Comment by dylanjw** [comment]
I broke out the commit re #1197. I will put it in a pull request once this is merged.

**Comment by dylanjw** [comment]
This is ready for review.

File: web3/manager.py (hunk 1 above)
Comment (kclowes) [reply web3/manager.py]: @dylanjw I am planning on taking this over shortly if you don't have objections! When I was looking this over, it looks like the attrdict_middleware didn't get moved over. I assume that the plan is to shift it still? Or did you run into something where it didn't make sense any more?

Comment (dylanjw) [reply web3/manager.py]: The plan was to move the attrdict_middleware functionality to the method formatters later.  I didnt run into any complications.

**Comment by kclowes** [comment]
Moved these commits over to #1461. Closing here.

**Pull request closed without merging**